### ISSUE TITLE
Making a particular ViewController as frontViewController in SWRevealViewController

### ISSUE THREAD

**Issue by gabhisekdev** [John-Lluch/SWRevealViewController#551]
![screen shot 2016-02-17 at 11 27 11 am](https://cloud.githubusercontent.com/assets/15869731/13103184/205bc122-d57a-11e5-9d16-82fb3aa62bb8.png)

I am doing a project in which I have implemented **SWRevealViewController** to achieve slide-out menu & now I have got the following scenario -

_I already have a view controller in my story board whose class is set to SWRevealViewController but that is not inter connected to my CreateNotificationVC by any means in the storyboard. Its a separate VC. The VC is in the bottom right corner  shown in the screenshot _

1)CreateNotificationVC embedded in a navigation controller CreateNotificationNVC.

2)I have a TableViewController which I want to be my rearVC.

**storyBoardID -**

CreateNotificationNVC - "createNtfctnNVC" rearTableViewController - "rearTVC"

**Scenario -**

I want CreateNotificationVC to act as frontViewController when user moves to this VC & implement all type of functionalities of SWRevealViewController.

**Code Till Now-**

In my CreateNotificationVC -

```
let revealView = self.storyboard?.instantiateViewControllerWithIdentifier("RevealVC") as! SWRevealViewController //Here I am instantiating my already present RevealViewController in storyboard
        let nvc = self.storyboard?.instantiateViewControllerWithIdentifier("createNtfctnNVC") as! createNtfctnNVC//createNtfctnNVC is the storyBoardId of Nav Controller in which CreateNotificationVC is embedded
        let rvc = self.storyboard?.instantiateViewControllerWithIdentifier("rearTVC") as! SlideMenuTVC//Here I am instantiating the TblViewCntrllr which acts as rearVC of my already present RevealViewController

        revealView.setFrontViewController(nvc, animated: true)
        revealView.pushFrontViewController(nvc, animated: true)
        revealView.setRearViewController(rvc, animated: true)

        self.view.addGestureRecognizer(revealView.panGestureRecognizer())
```

Problem - Swiping or toggling from any button in CreateNotificationVC does not result in the show up of slide menu.

Please help about this as not much works for this 3rd Party is found in swift.

**Comment by iDevelopper** [comment]
Please have a look here : http://www.appcoda.com/sidebar-menu-swift/

**Comment by iDevelopper** [comment]
And #524

**Comment by gabhisekdev** [comment]
@iDevelopper  Please check the code in my question you can notice that I have created an instance of SWR in that? Appcoda tutorial doesn't have a scenario like this.

**Comment by iDevelopper** [comment]
How and where do you create your instance of your CreateNotificationVC?

Instead, you have to create an instance of SWRevealViewController at the same place and set rear and front but not in the controller itself. And push it with -pushFrontViewController:animated method.

In your controller, just add the gesture recognizer to your view et set target (self.revealViewController) and action (revealToggle:) on your button.

**Comment by gabhisekdev** [comment]
@iDevelopper  I have written the below mentioned code in the viewDidLoad() of CreateNotificationVC - 

```
 let revealView = self.storyboard?.instantiateViewControllerWithIdentifier("RevealVC") as! SWRevealViewController //Here I am instantiating my already present RevealViewController in storyboard
        let nvc = self.storyboard?.instantiateViewControllerWithIdentifier("createNtfctnNVC") as! createNtfctnNVC//createNtfctnNVC is the storyBoardId of Nav Controller in which CreateNotificationVC is embedded
        let rvc = self.storyboard?.instantiateViewControllerWithIdentifier("rearTVC") as! SlideMenuTVC//Here I am instantiating the TblViewCntrllr which acts as rearVC of my already present RevealViewController

        revealView.setFrontViewController(nvc, animated: true)
        revealView.pushFrontViewController(nvc, animated: true)
        revealView.setRearViewController(rvc, animated: true)

        self.view.addGestureRecognizer(revealView.panGestureRecognizer())

```

Now Please check,and if also by any means I could send you my dummy code.

**Comment by iDevelopper** [comment]
It is not the good place to do that. You have to do that where you create your CreateNotificationVC or in the prepareForSegue method. You can send me your project if you want.

**Comment by GABHISEKBUNTY** [comment]
@iDevelopper  please check the source code at - https://www.dropbox.com/s/02fjub838hv95cr/SWRevealVC.zip?dl=0

**Comment by iDevelopper** [comment]
What about this?

https://www.dropbox.com/sh/lywt7rjlejhkmkz/AAAxm-XQZTR79ZzKpYa-92tQa?dl=0

**Comment by GABHISEKBUNTY** [comment]
@ its a demo project. copy the link and download. I don't know why direct opening shows error.

**Comment by iDevelopper** [comment]
What about this?

https://www.dropbox.com/sh/lywt7rjlejhkmkz/AAAxm-XQZTR79ZzKpYa-92tQa?dl=0

**Comment by gabhisekdev** [comment]
[SWRevealVC.zip](https://github.com/John-Lluch/SWRevealViewController/files/134313/SWRevealVC.zip)
@iDevelopper Okay check the file attached. Sorry for the inconvinience.

**Comment by iDevelopper** [comment]
I had no error! Did you see my response?

**Comment by gabhisekdev** [comment]
@iDevelopper Yeah. I got the point exactly. Thanks a lot. And it would be quite helpful on my part if you could explain my mistakes. Please.

**Comment by iDevelopper** [comment]
Your mistake is : you create your view controller without SWR.

![2016-02-17_14-30-42](https://cloud.githubusercontent.com/assets/1782450/13111007/a656b8d2-d583-11e5-95c2-e47ec2e9f12d.png)
![2016-02-17_14-32-08](https://cloud.githubusercontent.com/assets/1782450/13111008/a683a6f8-d583-11e5-8410-de3cb385b80b.png)

**Comment by iDevelopper** [comment]
hope this help you! I don't know swift very well, i'am developing in objective-c!

**Comment by gabhisekdev** [comment]
@iDevelopper  For eg -  If a certain ViewController's view(VC-B) is to be added as childVC in a SuperViewController(VC-A) which is in SWR stack & you have a button in the childVC(VC-B) which connects you to another VC(VC-C). Can we have SWR functionality in the VC-C?

More appropriately,How to make the child VC to be in the SWR stack while presenting its view by adding it as subView in VC-A?

**Comment by iDevelopper** [comment]
Send me an other project about this, thank you

**Comment by gabhisekdev** [comment]
@iDevelopper Thanks a lot for your help to a novice like me. Here is the demo proj. with all the codings. You will see 3 VC's one is the initial Front VC on which NotificationVC's view is presented & has a button in it to move to the createNotificationVC. All I want is that user shall be able to see the slide out menu from the createNotificationVC. Can you please consier viewing this?

[SWRevealVC.zip](https://github.com/John-Lluch/SWRevealViewController/files/135645/SWRevealVC.zip)

**Comment by iDevelopper** [comment]
The first ViewController is useless? What is it for?

**Comment by gabhisekdev** [comment]
@iDevelopper  No I have a scenario in which I have a scroll view in the first VC/frontVC in which I add multiple view controller's view as a paging sequence... So I need to add their view's as a part of scrollView's content. So I just gave you a demo of that only i.e addition of the child VC's view.

**Comment by iDevelopper** [comment]
Ah i removed it!
[SWRevealVC-3.zip](https://github.com/John-Lluch/SWRevealViewController/files/135836/SWRevealVC-3.zip)

**Comment by gabhisekdev** [comment]
@iDevelopper  It would be really helpful if you could do something with the requirements remaining intact.

**Comment by iDevelopper** [comment]
Could you explain in detail your requirements?

**Comment by gabhisekdev** [comment]
@iDevelopper  Basically in my front VC I am having a paging scroll view In which I add view's of 3 ViewController's. As one of my VC is NotificationVC,which is to be presented as paging on the scroll view. It has a button to create Notification, which opens up another ViewController named as createNotificationViewController. My requirement is to show the sliding menu on all the screens starting from the firstVC to the createNotificationViewController.

**Comment by iDevelopper** [comment]
Paging on the scroll view: You mean a UIPageViewController?

**Comment by gabhisekdev** [comment]
No Its something like using HMSegmentedController for segments with attached scroll view for scrolling effect or paging effect. For which its not as UIPageViewController?

**Comment by iDevelopper** [comment]
Then your frontViewController is a HMSegmentedController?

**Comment by gabhisekdev** [comment]
@iDevelopper  my front VC is  a subclass of UIViewController &  HMSegmentedController is a 3rd party used to add a custom segment. I have tied up the scroll view to the customSegmentedControl to provide an Android like segmented control + swipe between view controller feature.

Basically on the scroll view I am putting the views of different view controllers. So my demo project sent is just a simple example of it. So the conditions are same ,as I have earlier mentioned. So can I have slide menu in my createNotificationViewController.

**Comment by iDevelopper** [comment]
Yes and each segment has a corresponding controller and one of these controller is notificationVC, exact?

**Comment by gabhisekdev** [comment]
@iDevelopper Please check this project,it has exactly what I meant,except I have not implemented the SWR in this project.
[SegmentedControll.zip](https://github.com/John-Lluch/SWRevealViewController/files/136023/SegmentedControll.zip)

**Comment by gabhisekdev** [comment]
@iDevelopper  Did you get the source code. Please reply if you have got.

**Comment by iDevelopper** [comment]
Yes, super. I will have a look this afternoon.

**Comment by gabhisekdev** [comment]
@iDevelopper  Please have a look & reply what can be done to achieve so!!

**Comment by gabhisekdev** [comment]
@iDevelopper  Did you check the code? I f so please reply on how to achieve my rqrmnt via SWR

**Comment by iDevelopper** [comment]
Yes : 

[SegmentedControll.zip](https://github.com/John-Lluch/SWRevealViewController/files/137527/SegmentedControll.zip)

**Comment by GABHISEKBUNTY** [comment]
@iDevelopper  But requirement is that how we can get the sliding Action from the create Notification page. Can you please check that. Even also I have done what you have did, But my requirement is how we can show sliding menu from createNotification View controller? Please check your codes a bit.

**Comment by GABHISEKBUNTY** [comment]
@iDevelopper  What changes did you make and how did the slide menu show from creatNotificationVC, when you have not connected it to SWR

**Comment by GABHISEKBUNTY** [comment]
@iDevelopper  Can the slide menu be shown from view controllers presented modally?

**Comment by gabhisekdev** [comment]
Thanks got...it..

But what mistake was I doing bro?

**Comment by iDevelopper** [comment]
I received some mails but messages are not on Github, did you delete them?

**Comment by gabhisekdev** [comment]
@iDevelopper  Yeah.....
 I got what I was doing wrong. So I deleted those..Please check my understanding  -

1.I just embedded the  createNotificationVC in a Navigation Controller because I had got to show custom navigation bar & when I connect the notificationVC to the createNotificationVC's navigation controller via a default show segue,you will get error & if u connect via custom SWRPushSegue,then I get it correct.
1. When shall we connect SWRevealViewControllerSeguePushController & normal show  segue.
   **please run my code & follow the instructions.**

[SegmentedControll.zip](https://github.com/John-Lluch/SWRevealViewController/files/137641/SegmentedControll.zip)

**Comment by iDevelopper** [comment]
[CreateNotificationVC.swift.zip](https://github.com/John-Lluch/SWRevealViewController/files/137669/CreateNotificationVC.swift.zip)

**Comment by gabhisekdev** [comment]
@iDevelopper I am doing exactly what you did but getting insane error - Please check this project and tell me my point of mistakes &  I think this will be my last doubt in this section & it is exactly the same as I asked in the previous qstn about the connection of navigation controller via default segue & custom segue of SWRevealViewControllerSeguePushController . Thanks.

[SWRevealVC.zip](https://github.com/John-Lluch/SWRevealViewController/files/137671/SWRevealVC.zip)

**Comment by gabhisekdev** [comment]
Okay so what is the difference between adding in viewDidLoad() & viewWillAppear()? And kindly please the last source code?

**Comment by iDevelopper** [comment]
When you push the newVC, you add the recognizer to his view. A same recognizer can be added only to ONE view. Then when back to notificationVC you have to add it again.

**Comment by gabhisekdev** [comment]
Correct.....Bro then what about the nav.Controller thing & defaultSegue vs custom segue? Hats off to ur efforts. Thanks for helping so much to a fresher as me .....
As this was my first project without any senior devs in my company, I was wondering what to do?.....Thanks

**Comment by iDevelopper** [comment]
In your last project createNotificationVC has no more connection with SWR because SWR is connected with the first navigation controller. Then use custom push to replace the front view controller.

**Comment by iDevelopper** [comment]
Like you did in the previous project

**Comment by gabhisekdev** [comment]
Okay bro...got that...Thanks ....That's it...
Bro can I have your mail id to contact you & be in touch with you?

**Comment by iDevelopper** [comment]
patrick.bodet4@wanadoo.fr

**Comment by gabhisekdev** [comment]
Thanks a lot......

you can be reached via this id.....?

**Comment by iDevelopper** [comment]
Tell me when you got it. I will delete the post after

**Comment by iDevelopper** [comment]
Can you send me a mail to have your address?

**Comment by gabhisekdev** [comment]
@iDevelopper 

I am having an issue with SWRevealViewController again as -

1)When I click on create Notification button in my second tab, I am being able to move tyo the createNotificationVC but with no animation. SWR's animation is not working?

2)Clicking on back button should result in moving to frontVC with the same selected segment. I am using notificationPassing for that. But I think due to SWR, my notification observing is not working. 

Thanks.

[SegmentedControll.zip](https://github.com/John-Lluch/SWRevealViewController/files/150669/SegmentedControll.zip)
